### PR TITLE
tidb-in-kubernetes: expose services

### PR DESCRIPTION
**Pull request by cofyc**
https://github.com/cofyc/docs-cn/blob/expose-service/tidb-in-kubernetes/expose-servics.md

### CODE REVIEW

File: tidb-in-kubernetes/expose-servics.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+## NodePort
+
+在没有 LoadBalancer 时，可选择通过 NodePort 暴露。NodePort 有两种模式：
+
+- `externalTrafficPolicy=Cluster`: 集群所有的机器都会给 TiDB 分配 TCP
+  端口，此为默认值
+- `externalTrafficPolicy=Local`: 只有运行 TiDB 的机器会分配 TCP 端口，用于访问本地的 TiDB 实例
+
+使用 Local 模式时，建议打开 tidb-scheduler 的 `StableScheduling` 特性。tidb-scheduler 会尽可能在升级过程中将新 TiDB 实例调度到原机器，这样集群外客户端不用在 TiDB 重启后更新配置。
```
Comment (aylei) [inline tidb-in-kubernetes/expose-servics.md]: Maybe we should document or reference how to enable `StableScheduling`

File: tidb-in-kubernetes/expose-servics.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+# 服务暴露
+
+Kubernetes 集群内访问 TiDB 使用 TiDB service 域名 `<tidbcluster-name>.<namespace>` 即可。若需要集群外访问，则需将 TiDB 服务端口暴露出去。在 tidb-cluster helm chart 中，通过 values 文件中 `tidb.service` 字段配置：
+
+```yaml
+service:
+  type: NodePort
+  # externalTrafficPolicy: Cluster
+  # annotations:
+  # cloud.google.com/load-balancer-type: Internal
```
Comment (aylei) [inline tidb-in-kubernetes/expose-servics.md]: I'd recommend writing the full path:
```
tidb:
  service:
    type: NodePort
```
This code will just work by copy-and-paste

File: tidb-in-kubernetes/expose-servics.md (hunk 1 above)
Comment (cofyc) [reply tidb-in-kubernetes/expose-servics.md]: Do we have a doc which contains all configuration options of chart's values? I think we can add a link to `tidb-scheduler` configuration section.

File: tidb-in-kubernetes/expose-servics.md (hunk 2 above)
Comment (cofyc) [reply tidb-in-kubernetes/expose-servics.md]: Updated.

**Comment by cofyc** [comment]
@lilin90 Could you take a look at this?

File: tidb-in-kubernetes/expose-servics.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
+---
+title: 服务暴露
+category: how-to
+---
+
+# 服务暴露
+
+Kubernetes 集群内访问 TiDB 使用 TiDB service 域名 `<tidbcluster-name>.<namespace>` 即可。若需要集群外访问，则需将 TiDB 服务端口暴露出去。在 tidb-cluster helm chart 中，通过 values 文件中 `tidb.service` 字段配置：
```
Comment (tennix) [inline tidb-in-kubernetes/expose-servics.md]: `<tidbcluster-name>-tidb.<namespace>`

File: tidb-in-kubernetes/expose-servics.md [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    # externalTrafficPolicy: Cluster
+    # annotations:
+    # cloud.google.com/load-balancer-type: Internal
+```
+
+## NodePort
+
+在没有 LoadBalancer 时，可选择通过 NodePort 暴露。NodePort 有两种模式：
+
+- `externalTrafficPolicy=Cluster`: 集群所有的机器都会给 TiDB 分配 TCP
```
Comment (yikeke) [inline tidb-in-kubernetes/expose-servics.md]:
```suggestion
- `externalTrafficPolicy=Cluster`：集群所有的机器都会给 TiDB 分配 TCP
```

File: tidb-in-kubernetes/expose-servics.md [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+  端口，此为默认值
+- `externalTrafficPolicy=Local`: 只有运行 TiDB 的机器会分配 TCP 端口，用于访问本地的 TiDB 实例
```
Comment (yikeke) [inline tidb-in-kubernetes/expose-servics.md]:
```suggestion
- `externalTrafficPolicy=Local`：只有运行 TiDB 的机器会分配 TCP 端口，用于访问本地的 TiDB 实例
```

File: tidb-in-kubernetes/expose-servics.md (hunk 3 above)
Comment (cofyc) [reply tidb-in-kubernetes/expose-servics.md]: fixed

File: tidb-in-kubernetes/expose-servics.md [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+
+使用 Local 模式时，建议打开 tidb-scheduler 的 `StableScheduling` 特性。tidb-scheduler 会尽可能在升级过程中将新 TiDB 实例调度到原机器，这样集群外客户端不用在 TiDB 重启后更新配置。
+
+### 查看 NodePort 模式下对外暴露的 IP/PORT
+
+查看 Service 分配的 Node Port，可通过 kubectl 获取 TiDB 的 service 对象获知。
```
Comment (yikeke) [inline tidb-in-kubernetes/expose-servics.md]:
```suggestion
查看 Service 分配的 Node Port，可通过 kubectl 获取 TiDB 的 service 对象获知。
```
Is ”对象获知“ a typo?

Comment (cofyc) [reply tidb-in-kubernetes/expose-servics.md]: not a typo, but maybe "可通过 TiDB 的 service 对象来获知" is better.

Comment (cofyc) [reply tidb-in-kubernetes/expose-servics.md]: updated

Comment (yikeke) [reply tidb-in-kubernetes/expose-servics.md]: Yes, thanks for your clarification.

**Comment by yikeke** [comment]
**Note:** I have updated all code blocks according to our new code block format rules, please see [Update code block format to better support copy](https://internal.pingcap.net/jira/browse/ICM-95) for details. 

PTAL. @cofyc

**Comment by cofyc** [comment]
@yikeke hi, could you update again? I guess I have overwritten your commit by accident.

**Comment by yikeke** [comment]
> @yikeke hi, could you update again? I guess I have overwritten your commit by accident.

No problem.

**Comment by cofyc** [comment]
Thanks!

File: tidb-in-kubernetes/expose-servics.md [hunk 7]
```diff
@@ -0,0 +1,74 @@
+---
+title: 服务暴露
+category: how-to
+---
+
+# 服务暴露
+
+Kubernetes 集群内访问 TiDB 使用 TiDB service 域名 `<tidbcluster-name>-tidb.<namespace>` 即可。若需要集群外访问，则需将 TiDB 服务端口暴露出去。在 tidb-cluster helm chart 中，通过 values 文件中 `tidb.service` 字段配置：
```
Comment (yikeke) [inline tidb-in-kubernetes/expose-servics.md]:
```suggestion
在 Kubernetes 集群内访问 TiDB 时，使用 TiDB service 域名 `<tidbcluster-name>-tidb.<namespace>` 即可。若需要在集群外访问，则需将 TiDB 服务端口暴露出去。在 tidb-cluster helm chart 中，通过 values 文件中的 `tidb.service` 字段进行配置：
```

File: tidb-in-kubernetes/expose-servics.md [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+    # annotations:
+    # cloud.google.com/load-balancer-type: Internal
+```
+
+## NodePort
+
+在没有 LoadBalancer 时，可选择通过 NodePort 暴露。NodePort 有两种模式：
+
+- `externalTrafficPolicy=Cluster`：集群所有的机器都会给 TiDB 分配 TCP
+  端口，此为默认值
```
Comment (yikeke) [inline tidb-in-kubernetes/expose-servics.md]:
```suggestion
```

File: tidb-in-kubernetes/expose-servics.md [hunk 9]
```diff
@@ -0,0 +1,74 @@
...
+    # externalTrafficPolicy: Cluster
+    # annotations:
+    # cloud.google.com/load-balancer-type: Internal
+```
+
+## NodePort
+
+在没有 LoadBalancer 时，可选择通过 NodePort 暴露。NodePort 有两种模式：
+
+- `externalTrafficPolicy=Cluster`：集群所有的机器都会给 TiDB 分配 TCP
```
Comment (yikeke) [inline tidb-in-kubernetes/expose-servics.md]:
```suggestion
- `externalTrafficPolicy=Cluster`：集群所有的机器都会给 TiDB 分配 TCP 端口，此为默认值
```

File: tidb-in-kubernetes/expose-servics.md [hunk 10]
```diff
@@ -0,0 +1,74 @@
...
+  端口，此为默认值
+- `externalTrafficPolicy=Local`：只有运行 TiDB 的机器会分配 TCP 端口，用于访问本地的 TiDB 实例
+
+使用 Local 模式时，建议打开 tidb-scheduler 的 `StableScheduling` 特性。tidb-scheduler 会尽可能在升级过程中将新 TiDB 实例调度到原机器，这样集群外客户端不用在 TiDB 重启后更新配置。
```
Comment (yikeke) [inline tidb-in-kubernetes/expose-servics.md]:
```suggestion
    使用 Local 模式时，建议打开 tidb-scheduler 的 `StableScheduling` 特性。tidb-scheduler 会尽可能在升级过程中将新 TiDB 实例调度到原机器，这样集群外的客户端便不需要在 TiDB 重启后更新配置。
```

File: tidb-in-kubernetes/expose-servics.md [hunk 11]
```diff
@@ -0,0 +1,74 @@
...
+    
+    {{< copyable "shell-regular" >}}
+
+    ```shell
+    kubectl -n stability-cluster1 get pods -l "app.kubernetes.io/component=tidb,app.kubernetes.io/instance=${release}" -ojsonpath="{range .items[*]}{.spec.nodeName}{'\n'}{end}"
+    ```
+
+## LoadBalancer
+
+若运行在有 LoadBalancer 的环境，比如 GCP/AWS 平台，建议使用云平台 LoadBalancer 特性。
```
Comment (yikeke) [inline tidb-in-kubernetes/expose-servics.md]:
```suggestion
若运行在有 LoadBalancer 的环境，比如 GCP/AWS 平台，建议使用云平台的 LoadBalancer 特性。
```

**Comment by cofyc** [comment]
Thanks for reviewing!

@aylei I am merging this now. We can add a link to the relevant doc (e.g. helm chart how to) for `StableScheduling` feature later.

**Pull request merged**